skqulacs/qnn/solver.py

```diff
@@ -140,3 +140,27 @@ def run(
         loss = cost_func(theta_now, x, y)
         theta_opt = theta_now
         return loss, theta_opt
+
+
+@dataclass
+class Grad_Descent(Solver):
+    def run(
+        self,
+        cost_func: CostFunc,
+        jac: Jacobian,
+        theta: List[float],
+        x: NDArray[np.float_],
+        y: NDArray[np.float_],
+        lr: 0.1,
```

Comment:
For simple gradient descent, learning rate is not changed in iterations. So the `lr` should be a member variable, I believe.
For example, [`SGD`](https://pytorch.org/docs/stable/generated/torch.optim.SGD.html#torch.optim.SGD) in PyTorch is initialized with the learning rate.

Reply:
I used `lr: float = 0.1,` now, not sure if it's what you meant :)

Reply:
ah, I also changed the name of `maxiter` in the fit function, because it can be `maxiter_or_lr`